crates/bevy_app/src/app.rs

```diff
@@ -1178,4 +1184,31 @@ mod tests {
         // it's nice they're so small let's keep it that way.
         assert_eq!(mem::size_of::<AppExit>(), mem::size_of::<u8>());
     }
+
+    #[test]
+    fn initializing_resources_from_world() {
```

Comment:
Other things to check like making sure that the resources are not overwritten, already happen in the tests for `World`.